### ISSUE TITLE
Call on contact click in Contacts tab

### ISSUE THREAD

**Issue by ksz16** [FossifyOrg/Phone#80]
### Checklist

- [X] I made sure that there are **no existing issues** - [open](https://github.com/FossifyOrg/Phone/issues) or [closed](https://github.com/FossifyOrg/Phone/issues?q=is%3Aissue+is%3Aclosed) - to which I could contribute my information.
- [X] I made sure that there are **no existing discussions** - [open](https://github.com/FossifyOrg/Phone/discussions) or [closed](https://github.com/FossifyOrg/Phone/discussions?discussions_q=is%3Aclosed) - to which I could contribute my information.
- [X] I have read the FAQs inside the app (Menu -> About -> FAQs) and my problem isn't listed.
- [X] **I have taken the time to fill in all the required details. I understand that the bug report will be dismissed otherwise.**
- [X] This issue contains only one feature request.
- [X] I have read and understood the [contribution guidelines](https://github.com/FossifyOrg/Phone/blob/master/CONTRIBUTING.md).
- [X] I optionally [donated](https://fossify.org/donate) to support the Fossify mission.

### Feature description

There are three "On contact click" options in the Fossify Contacts app: 
- Call contact
- View contact details
- Edit contact
It would be nice if a similar choice also existed in the Fossify Phone app. Currently, you can only call on click in the Favorites and Call History tabs, while the default behavior in the Contacts tab is view contact details.

### Why do you want this feature?

This will make the use of the application seamless.

**Comment by Aga-C** [comment]
I think it will be enough if after tapping on a contact, it will call it instead of opening Contacts app. All other options are available after long-press on a contact.

**Comment by ksz16** [comment]
I definitely agree with you.
I would prefer the Fossify Phone app to be more focused on the call making stuff, while the Fossify Contacts app handles the editing of entries. The long-press option currently has all the features needed.

**Comment by vertigo220** [comment]
I like how some apps handle this where tapping the name calls and tapping the thumbnail either shows a menu (with options to call, text, view contacts, etc) or just opens the contacts details (and ideally which it does should be an option). IMO it should do this on both the contacts and favorites tab, so if you want to be able to just tap to call, you can, but you can also tap to perform other functions.

**Comment by naveensingh** [comment]
I believe the usual behavior in phone apps is to just launch the contact details page. Google Phone does this. Some people might even be surprised if clicking a contact initiated a call instead of showing more options.

I like how Samsung Phone expands the tapped contact to show options like Call, Message, or View contact details. I'll probably be making some UI changes to the apps as time goes on and this could be part of that but for now, we can do either of these:
 - A preference in the settings with "View contact details" still being the default option.
 - Add another menu option named "Call".

**Comment by ksz16** [comment]
For me the biggest problem is that I do not have quick access to call contacts that are not favorites. To make a call to such a contact I have to change the tab, search for the contact, open its details and only then I can make a call. 

A solution to the problem could be to change method of searching in the Favorites tab. Currently in this tab the search only takes place within contacts that have been marked as favorites. In other apps in this tab you get search results covering your entire list of contacts. This allows you to have quick access to make a call from the Favorites tab as well.

Alternatively, solve it the way @vertigo220 writes about: tapping the name calls, tapping the thumbnail shows the details.

**Comment by inson1** [comment]
> I would prefer the Fossify Phone app to be more focused on the call making stuff, while the Fossify Contacts app handles the editing of entries. The long-press option currently has all the features needed.

this goes against my use case, as I have only phone app on my home screen because of call history and I can do everything from it, but more option for the the click is good idea

**Comment by ksz16** [comment]
I noticed that my idea was not well received. :smile:
I understand the arguments against it and therefore think the best solution is to leave the current settings.

In my usage scenario, the problem could be solved by changing the way contacts are searched in the app. Currently, the search in each tab is done only within that tab, e.g. in the Favorites tab only favorite contacts are searched. It would be better if the search regardless of the tab was done using the list of all contacts. This is exactly how it is solved in the AOSP Dialer - you search for a contact and clicking on the search result immediately connects to it.

Adding such a feature could reconcile the opinions of everyone involved in the discussion and make the use of the application seamless.

**Comment by vertigo220** [comment]
I agree searching on the favorites tab should show all results, and just list any results that are favorites at the top.

Yes, long-pressing shows all the various actions, allowing you to then access the contact details or send a text. However, there are three issues with this:

- I've used many dialers, and a common UI design is as I described above, enough that I consider it almost a standard and it's just what I expect a dialer to behave like, and so when one doesn't I find it less usable and intuitive
- Long-pressing can at times be more difficult, such as when driving
- The actions are at the top of the screen, which requires much more moving around the screen, and on large phones (which, let's face it, is just about all phones anymore) requires a bit of a reach. Being able to tap the thumbnail and/or swipe would make it easier to perform, especially one-handed. Another option would be tapping the thumbnail could show a pop-up next to it with options to text and view the contact details, as well as all numbers listed under that contact. That way, all this these actions, including being able to call any of the numbers, would be right there.

And this could simply be disabled through the options if users don't want to use it, though given that it would only occur when tapping on the thumbnail, I suspect most users that wouldn't use it wouldn't notice or care either way.

**Comment by naveensingh** [comment]
I decided **not** to make this change optional because, apart from the options menu, one can now also tap the contact photo to access details. This can be reconsidered if it bothers too many people, but until then, tapping a contact will initiate a call.

**Comment by FoxMcloud5655** [comment]
I very much dislike this change.  Please, please make this optional.  The fact that it used to work a different way means that a very much non-clear UX change now initiates actions I didn't want to take.  I don't enjoy randomly calling someone because I wanted to look at their contact info.

**Comment by Aga-C** [comment]
>  I don't enjoy randomly calling someone because I wanted to look at their contact info.

You can turn on the call confirmation dialog in Settings to prevent accidental calls.

**Comment by FoxMcloud5655** [comment]
It doesn't matter what setting I can change that will apply a workaround to fix a problem created by changing the UX without telling the user.

I did read the changelog, and no accidental calls actually took place.  That does not change the fact that a way to make this feature optional was considered and dismissed as unnecessary.  I am here to voice my complaints because I want the option to toggle it back to the original functionality.

UX changes are always tricky to handle, but not telling the user intuitively through the app that something fundamental has changed isn't a good thing.  More choices are always better for the end user, especially since more often than not, power users are the ones that choose to install apps just like this one.

**Comment by naveensingh** [comment]
Please file a new [feature request](https://github.com/FossifyOrg/Phone/issues/new?template=feature_request.yml) about making this optional (as suggested in the original [description above](https://github.com/FossifyOrg/Phone/issues/80#issue-2105036961)). It'll be considered if more people show support for it.

Note that, if added, this option will also apply to the favorites tab.
